src/Kerbero.Infrastructure/Clients/NukiHttpClient.cs

```diff
@@ -0,0 +1,70 @@
+using Flurl;
+using Flurl.Http;
+using Kerbero.Common.Entities;
+using Kerbero.Common.Exceptions;
+using Kerbero.Infrastructure.Options;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Kerbero.Infrastructure.Clients;
+
+public class NukiHttpClient
+{
+	private readonly NukiClientOptions _options;
+
+	public NukiHttpClient(IOptions<NukiClientOptions> options)
+	{
+		_options = options.Value;
+	}
+
+	/// <summary>
+	/// Build the Uri for the redirect call to Nuki for the code
+	/// </summary>
+	/// <param name="clientId"></param>
+	/// <returns />
+	public Uri AskCodeAndGetUriForRedirect(string clientId)
+	{
+		if (string.IsNullOrEmpty(clientId)) throw new InvalidClientIdException(clientId, 1);
+		var redirectUriClientId = $"{_options.MainDomain}"
+			.AppendPathSegment(_options.RedirectUriForCode)
+			.AppendPathSegment(clientId);
+		return $"{_options.BaseUrl}"
+			.AppendPathSegments("oauth", "authorize")
+			.SetQueryParams(new
+			{
+				response_type = "code",
+				client_id = clientId,
+				redirect_uri = redirectUriClientId.ToString(),
+				scope = _options.Scopes,
+			})
+			.ToUri();
+	}
+
+	public async Task<NukiAccountEntity> RetrieveTokensFromAuthenticationCode(string clientId, string authCode)
+	{
+		if (string.IsNullOrWhiteSpace(clientId)) throw new InvalidClientIdException(clientId, 1);
+		var redirectUriClientId = $"{_options.MainDomain}"
+			.AppendPathSegment(_options.RedirectUriForCode)
+			.AppendPathSegment(clientId);
+		var response = await $"{_options.BaseUrl}"
+		  .AppendPathSegment("oauth")
+		  .AppendPathSegment("token")
+		  .PostJsonAsync(new
+		  {
+		    client_id = clientId,
+		    client_secret = _options.ClientSecret,
+		    grant_type = "authorization_code",
+		    code = authCode,
+		    redirect_uri = redirectUriClientId,
+		  }).ReceiveString(); // default Json converter not working
+		var deserializeObject = JsonConvert.DeserializeObject<dynamic>(response);
+		return new NukiAccountEntity()
+		{
+			Token = deserializeObject!.access_token,
```

Comment:
Avoid unwrapping like this, it may throw a null pointer. Rather check for null before trying to use the `deserializeObject `